data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -97,8 +102,10 @@ private TableMigrationUtil() {
               .collect(Collectors.toList());
       DataFile[] datafiles = new DataFile[fileStatus.size()];
       Tasks.Builder<Integer> task = Tasks.range(fileStatus.size())
-              .stopOnFailure()
-              .throwFailureWhenFinished();
+          .throwFailureWhenFinished(!skipCorruptFiles);
+      if (!skipCorruptFiles) {
+        task.stopOnFailure();
```

Comment:
Nit: Instead of updating the `throwFailureWhenFinished` and then having this `if` statement, it might be cleaner to simply place the `throwFailureWhenFinished` inside of the if statement as well.